presto-orc/src/main/java/com/facebook/presto/orc/metadata/statistics/DateStatistics.java

```diff
@@ -24,26 +26,51 @@
     // 1 byte to denote if null + 4 bytes for the value (date is of integer type)
     public static final long DATE_VALUE_BYTES = Byte.BYTES + Integer.BYTES;
 
-    private final Integer minimum;
-    private final Integer maximum;
+    private static final int INSTANCE_SIZE = ClassLayout.parseClass(DateStatistics.class).instanceSize();
+    private static final byte MIN_MASK = 1 << 1;
+    private static final byte MAX_MASK = 1;
+
+    private final int minimum;
+    private final int maximum;
+
+    private byte hasMinMax;
```

Comment:
I'm not sure saving a single byte is worth the complexity of the masking solution.  If you really want to stick with the masking, add methods for all of the bit manipulations to make the code more readable.